conan/tools/files/conandata.py

```diff
@@ -11,8 +11,6 @@ def update_conandata(conanfile, data):
     this only works for updating the conandata on the export() method, it seems it would
     be plain wrong to try to change it anywhere else
     """
-    if not hasattr(conanfile, "export_folder") or conanfile.export_folder is None:
```

Comment:
This, we can still keep the ``if conanfile.export_folder is None``, if we invalidate it just after the ``export()`` method, can't we?

Reply:
I think it is not that easy, currently (in this PR), the `conanfile.export_folder` is available in many other places, because [this](https://github.com/conan-io/conan/pull/11055/files#diff-c2ca7643fc907db00b41a346805bc9926f9a77adeaefebb10f42d5c528544536R107-R108) and [this](https://github.com/conan-io/conan/pull/11055/files#diff-c15f4aacf65aa58b31a041bac63e4dd7337355d9ec44a8f32b2a5815d5eb5ea9R834-R835) because we want to read them pointing to the "base source", so we don't know if it is set because you are in the "export" method or not.

Reply:
I don't fully get it. ``conanfile.export_folder`` is defined just before calling ``conanfile.export()`` method, and removed after it. Not used anywhere else. The ``export_folder`` shouldnt be pointing to anywhere else in any other method, because it is only relevant to the recipe, and the recipe is no longer copied to the "source" folder in the cache.  What am I missing?